### ISSUE TITLE
Android 4.4 supported? [Update: Is SELinux the culprit?]

### ISSUE THREAD

**Issue by tagline15** [koush/Superuser#253]
Hi,

I have no troubles flashing the superuser.zip onto my OnePlus One (Android 4.4), but after I open the Superuser app nothing happens. No binary update, just nothing.
Root checker tells me that I'm not properly rooted. Is Android 4.4 the issue? I noticed that the zip file was last updated in February.
A helping hand would be kindly appreciated. :D

**Comment by FluorescentHallucinogen** [comment]
No, it is definitely not an Android 4.4 issue. I use CWM Superuser on my Google Nexus 5 with Android 4.4.4 and all is OK. Please note, that there are 2 versions of OnePlus One (codenamed Bacon): 1. Color OS (Chinese) edition and 2. CyanogenMod (International) edition. Both, Color OS and CyanogenMod are not vanilla Android (AOSP). CyanogenMod edition has a custom version of CyanogenMod, named 11S, based on regular CyanogenMod source code, but have additionally OTA support and access to Google Play services including Google Play Store. The latest versions of regular CyanogenMod builds already has ClockworkMod/Koush's Superuser app integrated, but CyanogenMod 11S isn't pre-rooted out of the box. With rooting of what edition of OnePlus One you have this trouble?

**Comment by tagline15** [comment]
I'm using the international edition with CyanogenMod 11S.
I noticed that you get a Superuser menu in settings when enabling the developer options by tapping the build number seven times. But this Superuser menu behaves like the Superuser app that I flashed (no binary update or something similar).
Thanks for looking into it.

**Comment by FluorescentHallucinogen** [comment]
You don't need manual installation with the recovery zip if you already have a built-in Superuser app embedded into system settings menu. Embedded Superuser app and su binary updating using OTA or CyanogenMod updater during updating CyanogenMod.

Are you have troubles with embedded Superuser app?

AFAIK, CyanogenMod 11S doesn't has embedded Superuser app. Correct me if I'm wrong.

> no binary update or something similar

The latest stable version of CWM Superuser is 1.0.3.0 dated February 2014. What update if you already use the latest version?

Is this issue closed?

**Comment by tagline15** [comment]
See here: https://forums.oneplus.net/threads/hidden-superuser-in-settings.61781/
The menu is there, but it has no function.

Like you said: CM11S doesn't come pre-rooted, so the binary won't be updated over the air.

I'm using the version as of February 2014. To make it clearer what I mean: When I flashed superuser on my old phone, it prompted me to update the binary or something like that, the first time I opened the app. And after that the root access worked.
When I do the same on my One, nothing happens and the Root Checker app tells me I'm not properly rooted.

**Comment by tagline15** [comment]
Maybe SELinux is the problem? It's set to "strict" by default (and I can't change it).
Can superuser cope with this setting?

**Comment by FluorescentHallucinogen** [comment]
Duplicate of #246.

**Comment by FluorescentHallucinogen** [comment]
> When I flashed superuser on my old phone, it prompted me to update the binary or something like that, the first time I opened the app. And after that the root access worked.

When I flash CWM Superuser zip via recovery menu on my Nexus 5, CWM Superuser don't show any messages to update it and root access work properly (if I flash zip with latest version of CWM Superuser).

> Maybe SELinux is the problem?

Built-in Superuser app in CyanogenMod M9 work properly with enforcing SELinux status.

I guess (almost sure) that su binary compiled for embedded Superuser manager app is not compatible with su binary compiled for Superuser manager as a separate APK app (and vice versa).

This may be helpful: https://github.com/julian-klode/bacon-superuser

P.S. You also can install CyanogenMod M9 into your OnePlus One that pre-rooted, but has no OTA support and access to Google Play services out of the box. You can flash GApps and use CyanogenMod Updater instead of OTA.

**Comment by tagline15** [comment]
Thanks for the link. :D 
Here is the link to flashable zips by the same user:
https://forums.oneplus.net/threads/flashable-zip-enable-integrated-superuser.75511/

This same user here writes, that quote: "Superuser did not work for me, because the su daemon was not started, so I created a manual startup script for it. "

https://forums.oneplus.net/threads/enabling-root-access-superuser-in-cm11s.58553/#post-2694102

How could I start the su daemon?

**Comment by Kalevaxx** [comment]
I've installed application, su binary and init.d script on Android 4.4.2 (Samsung Galaxy S5 Mini)
Superuser doesn't work on this platform.
The problem is that su --daemon is not able to start from init.d script (for example I've added /system/xbin/su --daemon & to /etc/init.d/99Superuser).
But if I start /etc/init.d/99Superuser manually from root context=u:r:init:s0,(using SuperSU) all works fine.
SuperSu works well for this system (it's started from init.d script too). May be info from here http://su.chainfire.eu/ will be helpfull.

**Comment by Kethen** [comment]
We'll have to rig the install a little differently, then it will still works on android 4.4

For me, I tried to root a galaxy note 3 T210, but I simply don't want to use supersu. Inorder to get it working I did the following things:
1. Grabbed a busybox binary here https://busybox.net/downloads/binaries/1.21.1/busybox-armv7l
2. Unpacked my boot.img to mess with the ramdisk with tools here https://forum.xda-developers.com/showthread.php?t=2319018
3. Put that busybox binary into / of the ramdisk with permission 700 (renamed to just busybox)
4. Modify init.rc to support /system/init.d

> service initd-system /busybox run-parts /system/etc/init.d
>    class main
>    oneshot
>    seclabel u:r:init:s0

5. Repack that ramdisk and flash it
6. Installed http://download.clockworkmod.com/superuser/superuser.zip from recovery
7. (Through adb) Add /system/etc/init.d/99superuser with permission 700
> #!/system/bin/sh
> /system/xbin/su --daemon &

This will rig the daemon to start with context init and then daemon will still work (still supports kitkat I guess), but then we need better ways to rig the application. For my stock boot.img, init.rc will not execute install-recovery.sh so that's one thing.

**Comment by ildar** [comment]
Why don't you use https://f-droid.org/app/me.phh.superuser ? It is adapted
for SELinux
